plugins/package-managers/bower/src/main/kotlin/Bower.kt

```diff
@@ -168,24 +168,17 @@ private fun hasCompleteDependencies(info: PackageInfo, scopeName: String): Boole
     return dependencyKeys.containsAll(dependencyRefKeys)
 }
 
-private val PackageInfo.key: String?
-    get() {
-        // As non-null dependency keys are supposed to define an equivalence relation for parsing 'missing' nodes,
-        // only the name and version attributes can be used. Typically, those attributes should be not null
-        // however in particular for root projects the null case also happens.
-        val name = pkgMeta.name.orEmpty()
-        val version = pkgMeta.version.orEmpty()
-        return "$name:$version".takeUnless { name.isEmpty() || version.isEmpty() }
-    }
+private val PackageInfo.key: Any
+    get() = endpoint
 
-private fun getPackageInfosWithCompleteDependencies(info: PackageInfo): Map<String, PackageInfo> {
-    val result = mutableMapOf<String, PackageInfo>()
+private fun getPackageInfosWithCompleteDependencies(info: PackageInfo): Map<Any, PackageInfo> {
```

Comment:
Why is the map's key now `Any` instead of `Entrypoint`?

Reply:
It's not necessary to expose the type. Are you ok with it?

Reply:
Well, if it could be `Entrypoint`, I'd prefer to be explicit about this for type safety. Was there a particular reason to go with `Any`?

Reply:
I only wanted to express that it is `not null` and avoid the need for changing the calling code again, should the type of the key change again. This can probably also be addressed with a type alias.

I can also change the type to ` Endpoint`. What do you prefer?

Reply:
> should the type of the key change again.

While I don't know about your further refactoring plans (if any), I'd say that a further change of the key's type is very unlikely.

> I can also change the type to ` Endpoint`. What do you prefer?

I'd have a preference for explicitly using `Endpoint`. I can also live with a `typealias`, but if we assume this to be a one-time change (see my statement above), then changing it to `Endpoint` once is the same amount of work as changing it to a `typealias` once.